#5 add vehicles view

Degath opened the pull request.

Commit:
#5 add vehicles view, routes, vehicle-service, new-vehicle template

- fix imports / communication between modules
- delete excessive comments
- delete excessive field
- remove fleetId column from table
- remove unused method
- move styles to css
- add aliases
- reformat vehicle management component html
- fix tests

loziuu commented on src/app/app.module.ts:
```diff
@@ -1,18 +1,41 @@
-    CoreComponent
+    VehiclesManagementComponent,
+    NewVehicleFormComponent,
+    NewVehicleComponent
```
Nie można po prostu dociągnąć modułu, który udostępnia te komponenty?

Degath replied:
poprawiono :+1:

Degath replied:
poprawiono :+1:

loziuu commented on src/app/app.module.ts:
```diff
@@ -1,18 +1,41 @@
+import { NewVehicleFormComponent } from './shared/forms/new-vehicle-form/new-vehicle-form.component';
+import { NewVehicleComponent } from './feature/new-vehicle/new-vehicle.component';
+import { TableModule } from '../../node_modules/primeng/table';
+import {AccordionModule} from 'primeng/accordion';     //accordion and accordion tab
```
Dzięki za komenta, nie sposób się domyśleć XD

Degath replied:
Usunięto.

loziuu commented on src/app/core/core.module.ts:
```diff
@@ -1,9 +1,16 @@
+    HttpClientModule
+  ], 
+  providers: [
+    { provide: VehiclesService, useClass: VehiclesRestService },
```
Jeden provider w app.module, teraz drugi tutaj. What's going on?

Degath replied:
Usunięto ten w `app.module`, pozostał w `core`.

loziuu commented on src/app/core/vehicles/vehicles-rest.service.ts:
```diff
@@ -0,0 +1,22 @@
+  }
+
+  public addVehicle(vehicle: Vehicle): Observable<Vehicle> {
+    return this.http.post<Vehicle>(environment.api + '/vehicles', vehicle);
```
Przemapuj sobie Vehicle na VehicleDTO/VehicleRequest/NewVehicleRequest/Whatever. Gdy zmieni się REST API to będziesz musiał zmienić sobie logikę mapowania, bez zmiany twojego modelu. 

Przy GET można zrobić podobnie

loziuu commented on src/app/feature/vehicles-management/vehicles-management.component.html:
```diff
@@ -0,0 +1,24 @@
+      <ng-template pTemplate="body" let-vehicle let-columns="columns">
+        <tr >
+
+          <td style="max-width: 650px" *ngFor="let column of columns" [ngClass]="vehicle[column.field] >=0  ? 'is-number' : null">
```
Wyciągałbym wszelkie style do cssów, łatwiej to potem wsio zmieniać, bo wszystko masz w jednym miejscu.

loziuu commented on src/app/feature/vehicles-management/vehicles-management.component.html:
```diff
@@ -0,0 +1,24 @@
+  <h2>Vehicles list</h2>
+
+  <div class="mt-4">
+    <p-table [columns]="columns" [value]="vehicles" [paginator]="true" [rows]="10">
```
Myślę, że paginacja i sortowanie będzie robione w backendzie, ale na razie może zostać

loziuu commented on src/app/feature/vehicles-management/vehicles-management.component.spec.ts:
```diff
@@ -0,0 +1,30 @@
+
+  it('should create', () => {
+    expect(component).toBeTruthy();
+  });
```
Testy też mógłbyś się nauczyć pisać XD

Degath replied:
Postaram się XD

loziuu commented on src/app/feature/vehicles-management/vehicles-management.component.ts:
```diff
@@ -0,0 +1,38 @@
+
+  vehicles: Vehicle[];
+  columns: any[];
+  abc: any;
```
defgh..... używasz tego gdzieś? xd

loziuu commented on src/app/shared/models/vehicle.ts:
```diff
@@ -0,0 +1,23 @@
+import { Transmission } from "./transmission";
+
+export class Vehicle {
+    // constructor(
```
ekhem

loziuu commented on src/app/app.module.ts:
```diff
@@ -1,18 +1,41 @@
-import { CoreComponent } from './core/core.component';
+import { VehiclesManagementComponent } from './feature/vehicles-management/vehicles-management.component';
+import { VehiclesService } from './core/vehicles/vehicles.service';
+import { VehiclesRestService } from './core/vehicles/vehicles-rest.service';
```
Skorzystaj z aliasowania. Pkt. 2 -> https://medium.com/@tomastrajan/6-best-practices-pro-tips-for-angular-cli-better-developer-experience-7b328bc9db81

loziuu commented on src/app/feature/vehicles-management/vehicles-management.component.ts:
```diff
@@ -0,0 +1,38 @@
+      { field: 'capacity', header: `Capacity` },
+      { field: 'power', header: `Power` },
+      { field: 'transmission', header: `Transmission` },
+      { field: 'fleetId', header: `Fleet id` },
```
Wywal te kolumne

loziuu reviewed (changes requested):
Trzymam kciuczki

loziuu commented on src/app/shared/dtos/vehicle-dto.ts:
```diff
@@ -0,0 +1,26 @@
+    ) { }
+
+
+    public static toVehicleDto(entity: Vehicle) {
```
jak już to `from` -> VehicleDto.from(vehicle);

Degath replied:
zmieniono.

loziuu commented on src/app/shared/dtos/vehicle-dto.ts:
```diff
@@ -0,0 +1,26 @@
+    }
+
+    public static toVehicle(dto: VehicleDto){
+        return new Vehicle(dto.aggregateId, dto.manufacturer, dto.model, dto.capacity, dto.power, dto.transmission, dto.fleetId, dto.productionYear);
```
A tu po co przekazywane DTO w parametrze? :O

Degath replied:
Usunąłem metodę.

loziuu commented on src/app/shared/models/vehicle.ts:
```diff
@@ -0,0 +1,13 @@
+import { Transmission } from "@app/shared/models/transmission";
+
+export class Vehicle {
```
Nie musisz nic zmieniać, tylko informacyjnie, ale tutaj nie musisz się już trzymać schemy, którą dostajesz z BE. Może sobie dzielić jak tylko chcesz :p

Degath replied:
a jak chce tak jak jest na BE? :slightly_frowning_face: 
(wezmę to pod uwagę na przyszłość) :smile:

loziuu commented on src/app/core/vehicles/vehicles-rest.service.ts:
```diff
@@ -0,0 +1,26 @@
+import { VehiclesService } from "@app/core/vehicles/vehicles.service";
+import { Injectable } from "@angular/core";
+import { Vehicle } from "@app/shared/models/vehicle";
+import { environment } from './../../../environments/environment';
```
`@env` ?

Degath replied:
dodano.

loziuu reviewed (commented):
takie małe bzdury

Commit:
#5 refactor

- add alias for environment import
- change name of for-each element
- remove redundant method from dto
